### DISCUSSION TITLE
Is it possible to integrate image.nvim with obsidian.nvim?

### DISCUSSION THREAD

**Discussion by Demianeen** [epwalsh/obsidian.nvim#294] [Ideas]
I recently found this awesome plugin! But I found myself unable to view the images in it. The solution that I found was a [image.nvim](https://github.com/3rd/image.nvim) plugin, but from what I see obsidian ignores all path to the image and just uses image name. Is it possible to somehow integrate them?(or any other plugin with such functionality)

**Comment by epwalsh** [comment]
Hey @Demianeen this is not on our roadmap, but if someone can find a clean way to integration it then I'd be happy to accept a PR!

**Comment by maxdulay** [comment]
The solution I have for this is to have all of my images in an assets/images folder which is defined both in Obsidian and obsidian.nvim and then in the image.nvim setup have something like this 
```lua
    markdown = {
      enabled = true,
      filetypes = { "markdown" }, 
	  resolve_image_path = function(document_path, image_path, fallback)
		image_path = "Path/to/vault" .. image_path

		return fallback(document_path, image_path)
	end,
    },
```

**Reply by fvnky07** [reply]
can you demonstrate how it works? Ive been looking for similar functionality!

**Reply by maxdulay** [reply]
https://github.com/user-attachments/assets/b62e2d6b-eb22-4f43-a1b8-4fcaf0cad913

**Reply by monkeyxite** [reply]
Nice solution, and how do you get the image_path with multi-workspace? Suppose need get the para from obsidian.nvim in someway? @maxdulay

**Reply by maxdulay** [reply]
If you do a `:help obsidian.Workspace` you'll probably find what you're looking for. I personally don't use workspaces.

**Reply by ficd0** [reply]
The issue I see with your solution is that it will return an Obsidian-formatted path for any image inside a markdown document it encounters, even if the markdown file isn't inside your vault & the image path is just a regular relative path.

**Comment by ficd0** [comment]
The following solution will work whether you open the file directly, such as `nvim /path/to/vault/note.md`, or if you are like me, and open Neovim in some other directory (usually `$HOME` in my case), and then use `ObsidianQuickSwitch` to open notes. It will *also* work in normal Markdown files anywhere else on your system. 

```Lua
-- inside 'lua/plugins/image.lua'
local function resolver(document_path, image_path, fallback)
  local vault_dir = "/path/to/your/vault"
  -- Check if the document is inside the vault
  if document_path:find(vault_dir, 1, true) then
    -- If so, return formatted path to image
    return vault_dir .. "/" .. image_path
  end
  -- fallback to default since not in vault
  return fallback(document_path, image_path)
end

return {
  {
    "3rd/image.nvim",
    opts = {
      integrations = {
        markdown = {
          resolve_image_path = resolver,
        },
      },
    },
  },
}
```
